Push notifications: Spike for Android (and partial iOS)

oskarth opened the pull request:
Progress for push notifications spike for Android. Related to https://github.com/status-im/status-react/issues/1599 and achieves (1) "ability to manually trigger a push notification." by exposing an FCMToken.

PR only for devs to tests, nothing user-facing here. Mostly sanity check, but welcome to check actual notifications too.

### Summary:
- Print FCMToken which can be used to send PNs from Firebase Console
- Include react-native-fcm and related packages
- Update capabilities for iOS
- google service config files for Android and iOS

### Steps to test:

iOS no crash test:
- Open Status on iOS, make sure it doesn't crash. If using device, note whether it asks for permissions or not.

Android emulator test:
- Open Status on Android, if using Android ensure you have google-play-services installed. Using Genymotion this can be achieved with 2.10 and Open Gapps installed (https://github.com/opengapps/opengapps) - if you have issues PM me.
- Run `adb logcat | grep token` and note token
- Put app in background and go to Firebase Console (https://firebase.google.com/docs/cloud-messaging/) to trigger notification to that specific device.

status: ready

yenda commented on modules/react-native-status/android/react-native-status.iml:
```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
this file is usually not commited

yenda commented on android/app/app.iml:
```diff
@@ -0,0 +1,290 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
same

yenda commented on android/StatusIm.iml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
same

yenda reviewed (commented):
3 of the files that you committed are generated by tooling, maybe we should add them to gitignore ?

oskarth commented:
@yenda done

alwx commented on src/status_im/ui/screens/events.cljs:
```diff
@@ -264,3 +266,11 @@
+ :get-fcm-token
+ (fn [_ _]
+   (notifications/get-fcm-token)
+   {}))
```
hm
why does it return `{}`?

oskarth replied:
@alwx just placeholder for effects to come - main thing is checking in progress so far, proving (1) in issue and making sure it doesn't break anything major

jeluard commented on android/app/build.gradle:
```diff
@@ -162,6 +162,10 @@ android {
 }
 
 dependencies {
+    compile project(':react-native-udp')
```
Why did this change?

oskarth replied:
See below

jeluard commented on android/app/src/main/java/im/status/ethereum/MainApplication.java:
```diff
@@ -43,6 +46,10 @@ public boolean getUseDeveloperSupport() {
         protected List<ReactPackage> getPackages() {
             List<ReactPackage> packages = new ArrayList<ReactPackage>(Arrays.asList(
                     new MainReactPackage(),
+                    new UdpSocketsModule(),
```
Is it related to firebase?

oskarth replied:
This and TCP module imports were autogenerated - I believe as a leftover from when I targeted API 25. Removed and seems to build fine.

jeluard commented on ios/StatusIm/StatusIm.entitlements:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
```
What is this file?

oskarth replied:
Necessary to turn on push notifications capabilities/entitlements as far as I can tell. Not present in any RN gitignore I've found. I believe it'll be fleshed out with production certificates.

jeluard replied:
:+1:

jeluard reviewed (commented):
Did you make changes to font files?

flexsurfer commented on ios/StatusIm/Info.plist:
```diff
@@ -1,91 +1,97 @@
+      <string>MaterialIcons.ttf</string>
+      <string>Octicons.ttf</string>
+      <string>Zocial.ttf</string>
+      <string>MaterialCommunityIcons.ttf</string>
```
why do we need to add new fonts?

oskarth replied:
These were autogenerated - I believe as a leftover from when I targeted API 25. Removed and seems to build fine.

oskarth committed:
Push notifications: Spike for Android

- Print FCMToken which can be used to send PNs from Firebase Console
- Include react-native-fcm and related packages
- Update capabilities for iOS
- google service config files for Android and iOS
